legacy/startup.S

```diff
@@ -20,7 +20,7 @@ memset_reg:
   .type reset_handler, STT_FUNC
 reset_handler:
 
-#if MEMORY_PROTECT
+#if PRODUCTION
```

Comment:
In future I'd really like to replace this instance of `PRODUCTION` for some other variable. Since it's the most confusing instance. You always want that `#if` block included unless using `PRODUCTION=1` bootloader and unsigned `PRODUCTION=1` firmware.

But I'll put that into another PR. It needs some though about how to do it correctly.